File: clients/eventodb-go/system_test.go
package eventodb

import (
	"context"
	"regexp"
	"testing"
)

func TestSYS001_GetServerVersion(t *testing.T) {
	tc := setupTest(t, "sys-001")
	ctx := context.Background()

	version, err := tc.client.SystemVersion(ctx)
	if err != nil {
		t.Fatalf("Failed to get version: %v", err)
	}

	if version == "" {
		t.Error("Expected non-empty version string")
	}

	// Verify it looks like a semver (e.g., "1.3.0")
	semverPattern := regexp.MustCompile(`^\d+\.\d+\.\d+`)
	if !semverPattern.MatchString(version) {
		t.Logf("Warning: version '%s' doesn't match semver pattern (may be dev version)", version)
	}

	t.Logf("Server version: %s", version)
}

func TestSYS002_GetServerHealth(t *testing.T) {
	tc := setupTest(t, "sys-002")
	ctx := context.Background()

	health, err := tc.client.SystemHealth(ctx)
	if err != nil {
		t.Fatalf("Failed to get health: %v", err)
	}

	if health.Status == "" {
		t.Error("Expected status to be set")
	}

	// Status should be "ok" or similar
	if health.Status != "ok" && health.Status != "healthy" {
		t.Logf("Warning: unexpected health status: %s", health.Status)
	}

	t.Logf("Server health: %s", health.Status)
}

File: clients/eventodb_ex/test/system_test.exs
defmodule EventodbEx.SystemTest do
  use ExUnit.Case, async: true
  import EventodbEx.TestHelper

  setup do
    {client, namespace_id, _token} = create_test_namespace("system")

    on_exit(fn ->
      cleanup_namespace(client, namespace_id)
    end)

    %{client: client}
  end

  test "SYS-001: Get server version", %{client: client} do
    assert {:ok, version, _client} = EventodbEx.system_version(client)
    assert is_binary(version)
    # Should match semver pattern (but allow "dev" for development builds)
    if !String.match?(version, ~r/^\d+\.\d+\.\d+/) do
      IO.puts("Warning: version '#{version}' doesn't match semver pattern (may be dev version)")
    end
  end

  test "SYS-002: Get server health", %{client: client} do
    assert {:ok, health, _client} = EventodbEx.system_health(client)
    assert is_map(health)
    assert health.status == "ok"
  end
end
